contrib/pylightning/lightning/lightning.py

```diff
@@ -804,18 +804,37 @@ def plugin_rescan(self):
         }
         return self.call("plugin", payload)
 
-    def sendpay(self, route, payment_hash, description=None, msatoshi=None):
-        """
-        Send along {route} in return for preimage of {payment_hash}
-        """
+    def _deprecated_sendpay(self, route, payment_hash, description=None, msatoshi=None):
+        warnings.warn("sendpay: the 'description' field is renamed 'label' : expect removal"
+                      " in early-2020",
+                      DeprecationWarning)
         payload = {
             "route": route,
             "payment_hash": payment_hash,
-            "description": description,
+            "label": description,
             "msatoshi": msatoshi,
         }
         return self.call("sendpay", payload)
 
+    def sendpay(self, route, payment_hash, *args, **kwargs):
+        """
+        Send along {route} in return for preimage of {payment_hash}
+        """
+
+        if 'description' in kwargs:
+            return self._deprecated_sendpay(route, payment_hash, *args, **kwargs)
+
+        def _sendpay(route, payment_hash, label=None, msatoshi=None):
```

Comment:
I find it strange that this function is a sub-function here, but `_deprecated_sendpay` is a method of the class.  Suggest making `_deprecated_sendpay` into a sub-function here as well.

Reply:
Good idea! Now I make it as a sub-function. I also make similar changes in #3073  .

Reply:
I think wrapping this in sub-functions is a bit strange. It's just a single `warning` and an assignment that differs.

Reply:
Yeah. Let this function be independent。